src/features/CastAndCrew/index.js

```diff
@@ -1,130 +1,93 @@
+import React, { useState, useEffect } from "react";
+import { getMovieCredits } from "../movieDetailsAPI"; // Upewnij się, że import jest poprawny
 import {
-  StyledPersonWrapper,
   Text,
+  StyledPersonWrapper,
   WrapperItem,
   ImageWrapper,
   WrapperActorName,
   WrapperRole,
-} from "./styled";
+} from "./styled"; // Upewnij się, że import jest poprawny
 import posterExample from "../../image/posterExample.png";
-import Crew from "../../image/Crew.png";
-import { useEffect, useState } from "react";
-import { getMovieCredits } from "../movieDetailsAPI";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchMovies, selectMovies, selectStatus } from "../movieBrowserSlice";
+
+export const CastAndCrew = ({ movieId }) => {
+  // const dispatch = useDispatch();
+
+  // const movies = useSelector(selectMovies);
+  // const status = useSelector(selectStatus);
+
+  // useEffect(() => {
+  //   dispatch(fetchMovies());
+  // }, [dispatch]);
 
-export const CastAndCrew = () => {
-  const [error, setError] = useState(false);
-  const [actor, setActor] = useState(null);
-  const movieId = 550; // może być przekazany jako parametr jeśli komponent ma dynamiczne ID
+  const [credits, setCredits] = useState({ cast: [], crew: [] });
+  // const [cast, setCast] = useState([]);
+  // const [crew, setCrew] = useState([]);
+
+  // const movieId = 550;
 
   useEffect(() => {
     const fetchMovieCredits = async () => {
       try {
-        const fetchedData = await getMovieCredits(movieId);
-        if (fetchedData) {
-          setActor(fetchedData.actors);
+        const fetchedCredits = await getMovieCredits(movieId);
+        console.log("Raw API response:", credits); // Sprawdź surową odpowiedź
+
+        if (fetchedCredits) {
+          console.log("Cast before setState:", credits.cast); // Sprawdź dane przed setState
+          setCredits(fetchedCredits);
         }
-      } catch (err) {
-        console.error("Błąd podczas pobierania szczegółów filmu:", err);
-        setError(true);
+      } catch (error) {
+        console.error("Error details:", error);
```

Comment:
Maybe when error occurs it should by now show Error Page component.

Reply:
Yes indeed, I will work on that.